### ISSUE TITLE
Move inpar files to related module

### ISSUE THREAD

**Issue by sebproell** [4C-multiphysics/4C#59]
_Ported from Gitlab. Original issue [here](https://gitlab.lrz.de/baci/baci/-/issues/962)_

## Description

`inpar_`-files should not exist on the top-level; the files should move into their respective physics modules. This allows to define the input parameters locally for each module itself and removes a central dependency `inpar` that pollutes every physics module.

## Possible Solution and Definition of Done

- [ ] `inpar` folder removed, all files moved to physics.

**Comment by davidrudlstorfer** [comment]
@bwirthl @lkoeglmeier I am currently moving inpar files again and I stumbled over the poro files again. One question:

We currently have the following poro files in inpar:

- 4C_inpar_poro.hpp (this just holds a random enum - the entire file can be deleted as far as I can tell, see #1007)
- 4C_inpar_poroelast.cpp/.hpp
- 4C_inpar_poroscatra.cpp/hpp

Do you have any preferences where to move the remaining poro inpar files? Or is this more complicated and needs to be split up?

**Comment by lkoeglmeier** [comment]
yes, the first file can be deleted - thanks! `4C_inpar_poroelast.cpp/.hpp` and `4C_inpar_poroscatra.cpp/hpp `belong to the pressure_velocity_based poro formulation. I would suggest moving them to the `poroelast `and `poroelast_scatra` folder, respectively?

**Comment by davidrudlstorfer** [comment]
@ppraegla Do you have any preferences/ideas on where I could/should put the particle input?

**Comment by ppraegla** [comment]
> [@ppraegla](https://github.com/ppraegla) Do you have any preferences/ideas on where I could/should put the particle input?

Ideally, this would be combined with #92. So, the particle input can be directly placed in the new `particle` folder. I have this on my agenda. But currently no time to work on it. I can look at it in two weeks.

**Comment by davidrudlstorfer** [comment]
So slowly this issue is coming to an end 👀 

The only files that remain are `io_runtime` files (8 files consisting of corresponding headers and implementations).

I currently do not have a real plan on where to put those files.

For example `4C_inpar_io.cpp` contains the general IO features and has a dependency on structure. But it also contains some top-level fluid parameters, binning and so on. Should this therefore be split up even more?

Maybe someone with more 4C knowledge (@4C-multiphysics/maintainer) could check this out. I would be open for any form of input (👀) and discussion.

**Comment by lauraengelhardt** [comment]
I just took a quick look but was not really involved in modularization ;) 

My opinion would be:
`4C_inpar_IO_runtime_output.hpp` to core/io.
`4C_inpar_IO_runtime_vtp_output_structure.hpp `and  `4C_inpar_IO_runtime_vtk_output_structure.hpp`: I would argue this belongs to `structure_new` (similar as you moved the beam vtk input). 
`4C_inpar_io.hpp` is very convoluted. Some features are also only working in certain physics (even if not marked e.g. "STRUCT" or so). I think we should split this: decide for each parameter if actually used; keep general IO features in IO; move physics dependent IO; ...

We can discuss on friday if you like.

**Comment by lauraengelhardt** [comment]
I would like to note here that RESTARTEVERY in the IO section does nothing for a solid simulation when RESTARTEVERY in the STRUCTURAL DYNAMICS section is not specified. 
It is used here: https://github.com/4C-multiphysics/4C/blob/d495b396eb2604d75de45535e8b9635cef958764/src/global_data/4C_global_data_read.cpp#L1367-L1371 and is apparently not a general restart but a restart after a certain walltime (and should be grouped with the RESTARTWALLTIMEINTERVAL parameter to avoid confusion. Afaik the walltime restart is not tested.
